samcli/local/events/api_event.py

```diff
@@ -234,3 +236,179 @@ def to_dict(self):
         }
 
         return json_dict
+
+
+class ContextHTTP:
+    def __init__(
+        self, method=None, path=None, protocol="HTTP/1.1", source_ip="127.0.0.1", user_agent="Custom User Agent String"
+    ):
+        """
+        Constructs a ContextHTTP
+
+        :param str method: HTTP Method for the request
+        :param str path: HTTP Path for the request
+        :param str protocol: HTTP Protocol for the request (Default: HTTP/1.1)
+        :param str source_ip: Source IP for the request (Default: 127.0.0.1)
+        :param str user_agent: User agent (Default: Custom User Agent String)
+        """
+        self.method = method
+        self.path = path
+        self.protocol = protocol
+        self.source_ip = source_ip
+        self.user_agent = user_agent
+
+    def to_dict(self):
+        """
+        Constructs an dictionary representation of the HTTP Object to be used
+        in serializing to JSON
+
+        :return: dict representing the object
+        """
+        json_dict = {
+            "method": self.method,
+            "path": self.path,
+            "protocol": self.protocol,
+            "sourceIp": self.source_ip,
+            "userAgent": self.user_agent,
+        }
+
+        return json_dict
+
+
+class RequestContextV2:
+    def __init__(
+        self,
+        account_id="123456789012",
+        api_id="1234567890",
+        http=None,
+        request_id="c6af9ac6-7b61-11e6-9a41-93e8deadbeef",
```

Comment:
Possibly legacy carry-over, but why always the same request ID here? Can we just generate a new valid request ID each time, in case someone is locally testing where they check for duplicated request IDs?

This shouldn't block launch if we're copying logic from REST API.